sequencer/src/header.rs

```diff
@@ -355,8 +368,7 @@ impl BlockHeader<SeqTypes> for Header {
             validated_state,
             instance_state.chain_config,
         )
-        // TODO we should be able to return an error from `Header::new`
-        .unwrap_or_else(|err| panic!("invalid proposal: {err:#}"))
+        .map_err(|e| InvalidBlockHeader::new(e.to_string()))
```

Comment:
For formatting anyhow errors, prefer `format!("{e:#}")` instead of `e.to_string()`. The former will print the whole chain of errors/contexts while the latter only prints the outermost